### PR TITLE
Feature/7648 family/7842 new card design

### PR DESCRIPTION
**Pull request by KaiTeuber**
## Description
Update card design to new layout with QR-Code and different gradient styles

## Link to Jira
https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-7842

## Screenshots
IMG_0017
IMG_0018

### CODE REVIEW

**Review by ArturFriesen — APPROVED** [review]

File: src/xcode/ENA/ENA/Resources/Localization/bg.lproj/Localizable.strings [hunk 1]
```diff
@@ -2451,23 +2451,11 @@
 
 "VaccinationCertificate_Registration_description" = "Добавете ваксинационните си сертификати в приложението, за да са винаги с вас. За целта сканирайте QR кода от вашия документ.";
 
-/* - Vaccination Certificate */
-"VaccinationCertificate_title" = "Електронно **Доказателство за ваксиниране**";
-
-"VaccinationCertificate_partiallyVaccinated" = "Ваксинирането не е пълно";
-
-"VaccinationCertificate_vaccinationValidUntil" = "Валидно до %@";
-
```
Comment (pascal-brause) [inline src/xcode/ENA/ENA/Resources/Localization/bg.lproj/Localizable.strings:2460]: I thought when removing keys in the German file, the keys will be deleted in the other languages files when ever the translation is re-done again. So we do not have to keep care of other files then the German one. Or do I am false?

File: src/xcode/ENA/ENA/Resources/Localization/de.lproj/Localizable.strings [hunk 2]
```diff
@@ -2606,26 +2606,18 @@ Die Angaben werden statistisch ausgewertet, sie werden nicht zu einem Profil ges
 
 "HealthCertificate_Overview_EmptyImageDescription" = "Zertifikatssiegel mit digitalem Zertifikat im Hintergrund";
 
-/* - Vaccination Certificate */
+"HealthCertificate_Overview_Covid_Title" = "EU Digitales";
 
-"VaccinationCertificate_title" = "EU Digitaler\n**COVID-Impfnachweis**";
+"HealthCertificate_Overview_Covid_Certificate" = "COVID-Zertifikat";
```
Comment (pascal-brause) [inline src/xcode/ENA/ENA/Resources/Localization/de.lproj/Localizable.strings:2611]: Old title for the DCC....See #2987

**Review by pascal-brause — CHANGES_REQUESTED** [review]
Request change due to the wrong, old, text for the title.

**Review by NataliaLemmerth — CHANGES_REQUESTED** [review]
Please change the certificate name to new official name: Digitales COVID-Zertifikat der EU

File: src/xcode/ENA/ENA/Source/Scenes/HealthCertificates/Overview/Cells/HealthCertifiedPerson/HealthCertifiedPersonCellModel.swift [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+	init(
+		testCertificate: HealthCertificate
+	) {
+		title = AppStrings.HealthCertificate.Overview.covidTitle
+		description = AppStrings.HealthCertificate.Overview.covidCertificate
+		backgroundGradientType = .lightBlue(withStars: true)
+		name = testCertificate.name.fullName
+		certificate = testCertificate
+		accessibilityIdentifier = AccessibilityIdentifiers.HealthCertificate.Overview.testCertificateRequestCell
+	}
```
Comment (nickguendling) [inline src/xcode/ENA/ENA/Source/Scenes/HealthCertificates/Overview/Cells/HealthCertifiedPerson/HealthCertifiedPersonCellModel.swift]: Should not be required anymore

File: src/xcode/ENA/ENA/Source/Scenes/HealthCertificates/Overview/Cells/HealthCertifiedPerson/HealthCertifiedPersonTableViewCell.swift [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+		nameLabel.text = cellModel.name
+		gradientView.type = cellModel.backgroundGradientType
+		// add a placeholder image and process the QRCode image in background to smooth scrolling a bit
+		qrCodeImageView.image = placeHolderImage
+		DispatchQueue.global(qos: .userInteractive).async {
+			let qrCodeImage = UIImage.qrCode(
+				with: cellModel.certificate.base45,
+				encoding: .utf8,
+				size: CGSize(width: 280, height: 280),
+				qrCodeErrorCorrectionLevel: .medium
```
Comment (nickguendling) [inline src/xcode/ENA/ENA/Source/Scenes/HealthCertificates/Overview/Cells/HealthCertifiedPerson/HealthCertifiedPersonTableViewCell.swift:42]: According to [the tech spec](https://github.com/corona-warn-app/cwa-app-tech-spec/blob/48b557b918634a6c87ef6de615734bb6613331c4/docs/spec/dgc-overview-client.md#generating-a-qr-code-for-a-dgc) (4.) we should use `.quartile` for digital COVID certificates

File: src/xcode/ENA/ENA/Resources/Localization/bg.lproj/Localizable.strings (hunk 1 above)
Comment (KaiTeuber) [reply src/xcode/ENA/ENA/Resources/Localization/bg.lproj/Localizable.strings:2460]: I made a rollback - thank for the notice

File: src/xcode/ENA/ENA/Source/Scenes/HealthCertificates/Overview/Cells/__tests__/HomeHealthCertifiedPersonCellModelTests.swift [hunk 5]
```diff
@@ -6,7 +6,7 @@ import XCTest
 import HealthCertificateToolkit
 @testable import ENA
 
-class HomeHealthCertifiedPersonCellModelTests: XCTestCase {
+class HealthCertifiedPersonCellModelTests: XCTestCase {
 
 	func testGIVEN_completelyProtectedCertifiedPerson_THEN_IsSetupCorrectly() throws {
```
Comment (nickguendling) [inline src/xcode/ENA/ENA/Source/Scenes/HealthCertificates/Overview/Cells/__tests__/HomeHealthCertifiedPersonCellModelTests.swift:11]: I think we only need to keep one of the tests as the vaccination state has no influence on the cell anymore

File: src/xcode/ENA/ENA/Source/View Helpers/AppStrings.swift [hunk 6]
```diff
@@ -1917,23 +1917,9 @@ enum AppStrings {
...
-				static let description = NSLocalizedString("TestCertificate_Info_description", comment: "")
-			}
-
-			enum TestCertificate {
-				static let title = NSLocalizedString("TestCertificate_title", comment: "")
-				static let testDate = NSLocalizedString("TestCertificate_testDate", comment: "")
-			}
+			static let covidTitle = NSLocalizedString("HealthCertificate_Overview_Covid_Title", comment: "")
+            static let covidCertificate = NSLocalizedString("HealthCertificate_Overview_Covid_Certificate", comment: "")
+			static let covidDescription = NSLocalizedString("HealthCertificate_Overview_Covid_Certificate_Description", comment: "")
```
Comment (nickguendling) [inline src/xcode/ENA/ENA/Source/View Helpers/AppStrings.swift]: Indentation looks off here, just a GitHub issue?

File: src/xcode/ENA/ENA/Resources/Localization/de.lproj/Localizable.strings [hunk 7]
```diff
@@ -2606,26 +2606,18 @@ Die Angaben werden statistisch ausgewertet, sie werden nicht zu einem Profil ges
 
 "HealthCertificate_Overview_EmptyImageDescription" = "Zertifikatssiegel mit digitalem Zertifikat im Hintergrund";
 
-/* - Vaccination Certificate */
+"HealthCertificate_Overview_Covid_Title" = "Digitales\n**COVID-Zertifikat der EU**";
 
-"VaccinationCertificate_title" = "EU Digitaler\n**COVID-Impfnachweis**";
+"HealthCertificate_Overview_Covid_Certificate" = "COVID-Zertifikat";
```
Comment (nickguendling) [inline src/xcode/ENA/ENA/Resources/Localization/de.lproj/Localizable.strings:2611]: Is this still needed?

File: src/xcode/ENA/ENA/Source/Views/CardView.swift [hunk 8]
```diff
@@ -8,9 +8,40 @@ class CardView: UIView {
...
-		super.awakeFromNib()
+	override init(frame: CGRect) {
+		super.init(frame: frame)
+		setup()
+	}
+
+	required init?(coder: NSCoder) {
+		super.init(coder: coder)
+		setup()
+	}
```
Comment (nickguendling) [inline src/xcode/ENA/ENA/Source/Views/CardView.swift:19]: Should go to a `// MARK: - Init` section

**Review by nickguendling — CHANGES_REQUESTED** [review]

File: src/xcode/ENA/ENA/Source/Scenes/HealthCertificates/Overview/Cells/HealthCertifiedPerson/HealthCertifiedPersonCellModel.swift (hunk 3 above)
Comment (KaiTeuber) [reply src/xcode/ENA/ENA/Source/Scenes/HealthCertificates/Overview/Cells/HealthCertifiedPerson/HealthCertifiedPersonCellModel.swift]: removed it with matching unittest

File: src/xcode/ENA/ENA/Source/Scenes/HealthCertificates/Overview/Cells/HealthCertifiedPerson/HealthCertifiedPersonTableViewCell.swift (hunk 4 above)
Comment (KaiTeuber) [reply src/xcode/ENA/ENA/Source/Scenes/HealthCertificates/Overview/Cells/HealthCertifiedPerson/HealthCertifiedPersonTableViewCell.swift:42]: done

File: src/xcode/ENA/ENA/Source/Scenes/HealthCertificates/Overview/Cells/__tests__/HomeHealthCertifiedPersonCellModelTests.swift (hunk 5 above)
Comment (KaiTeuber) [reply src/xcode/ENA/ENA/Source/Scenes/HealthCertificates/Overview/Cells/__tests__/HomeHealthCertifiedPersonCellModelTests.swift:11]: yeah, I removed the other two

File: src/xcode/ENA/ENA/Resources/Localization/de.lproj/Localizable.strings (hunk 7 above)
Comment (KaiTeuber) [reply src/xcode/ENA/ENA/Resources/Localization/de.lproj/Localizable.strings:2611]: no, I removed it

File: src/xcode/ENA/ENA/Source/Views/CardView.swift (hunk 8 above)
Comment (KaiTeuber) [reply src/xcode/ENA/ENA/Source/Views/CardView.swift:19]: 👍

**Review by pascal-brause — APPROVED** [review]

**Review by NataliaLemmerth — APPROVED** [review]
UA approved

**Review by nickguendling — APPROVED** [review]

**Pull request merged**